### ISSUE TITLE
Why is the reproduction result inferior to the paper?

### ISSUE THREAD

**Issue by Fgfly** [blacksingular/GHRN#1]
Hello, I have some confusion.
1.~Run: (1) ‘python main.py --dataset amazon --homo 0’, get>> (MF1 92.39 AUC 97.06),
Then run (2) ‘python main.py --dataset amazon --homo 0 --del_ratio 0.015‘,get>> (MF1 90.63 AUC 96.43), this seems to get worse，
Even if run (3) ‘python main.py --dataset amazon --homo 0 --del_ratio 0.015 --run 10‘， 
                       the average result of 10 runs is (MF1-mean: 91.54, MF1-std:0.56, AUC-mean: 96.93, AUC-std: 0.39), 
this is still a worse result. Can you help explain why?

2.In 'dataset.py', there are errors in lines 55 and 61, and flag = 1 is missing between 63 and 64, and flag += 1 is missing between 71 and 72. After fixing the code, I still get worse results than BWGNN in yelp dataset.

Could you tell me the correct way to reproduce the paper? thank you

**Comment by blacksingular** [comment]
Hi, there. Thanks for your interest in our work. For your reproduction questions, I'll help you.
1. We have multiple arguments in our code. The most important ones are 'load_epoch', 'del_ratio' and 'adj_type'. You can tune these arguments before running. For example, for amazon dataset, you can set 'load_epoch' as 40 and 'del_ratio' as 0.015. Another option is 'load_epoch' as 100 and 'del_ratio' as 0.02. 
2. Thanks for your suggestion. The code typos between lines 55 and 72 will be fixed soon.

**Comment by Fgfly** [comment]
Thank you for your reply, but I still haven't got the result in the paper.

1.‘python main.py --dataset amazon --homo 0’，get>>(**MF1 92.39 AUC 97.06**)，it‘s result of BWGMM；
2.‘python main.py --dataset amazon --homo 0 --load_epoch 40 --del_ratio 0.015’
    get>>(**MF1 91.06 AUC 96.88**)
3.'python main.py --dataset amazon --homo 0 --load_epoch 40 --del_ratio 0.015 --run 10'
    get>>(**MF1-mean: 91.47**, MF1-std: 0.54, **AUC-mean: 96.95**, AUC-std: 0.39)
4.'python main.py --dataset amazon --homo 0 --load_epoch 100 --del_ratio 0.02'
    get>>(**MF1 92.10 AUC 97.06**)
5.'python main.py --dataset amazon --homo 0 --load_epoch 100 --del_ratio 0.02 --run 10'
    get>>(**MF1-mean: 91.86**, MF1-std: 0.52, **AUC-mean: 96.92**, AUC-std: 0.45)

According to the super parameters you provided and the paper proposed that the optimal del_ratio is 0.015 and 0.02 in amazon dataset. However, the result of my experiment is still worse than that of BWGNN. Could you further answer my confusion?

**Comment by blacksingular** [comment]
Well, that's weird. I just ran the code this afternoon, and with the hyperparameter I provided, I got AUC with 97.18 and 97.31 respectively. I don't know the exact reason but I guess it maybe the CUDA difference or the environment difference. Our model is tested on GeForce RTX 3090 with CUDA 11.7, the python version is 3.9.12 and the pytorch version is 1.9.0. Hope that helps!

**Comment by Fgfly** [comment]
My device also contains GeForce RTX 3090, I put the environment consistent with yours, except dgl  0.9.1, but the results were pretty much the same.

And when I set '**python main.py --dataset amazon --homo 1**',，get>>(**MF1 91.94 AUC 93.95**), which is the same as the paper, so the cause is not environmental.

Could you tell me the result of F1-macro and the corresponding relationship with the configuration?I'll find a way to reproduce it. Thanks!

**Comment by blacksingular** [comment]
Sure. Below are the configurations and results. 

For 'load_epoch' 40 and 'del_ratio' 0.015, REC 80.69 PRE 91.20 MF1 92.12 AUC 97.18.
For 'load_epoch' 100 and 'del_ratio' 0.02, REC 82.55 PRE 89.23 MF1 92.18 AUC 97.31.

**Comment by Fgfly** [comment]
I became more confused, this result is obviously contrary to the conclusion in **Figure 7 (b)** of the paper.
The random seeds in the code are determined, and the results of your experiment will be exactly the same every time.

Is the result of Table 1 in the paper a specific random seed result?Can you tell me which random seed it is?
Thanks!
